Resources/doc/definitions/type-inheritance.md

```diff
@@ -0,0 +1,101 @@
+Inheritance
+===========
+
+In some cases, inheritance can helps to don't repeat your self, beater organisation,
+clearer schema, easier interfaces implementation...
+
+Here an example using inheritance:
+
+```yaml
+ObjectB:
+    type: object
+    config:
+        interfaces: [InterfaceA]
+        fields:
+             foo: {type: String!}
+
+# ObjectA inherited config from InterfaceA and ObjectC
+ObjectA:
+    type: object
+    # ObjectB inherited config (fields, args...) from ObjectA
+    hiers: [ObjectB]
+    extends: [InterfaceA, ObjectC]
+
+ObjectC:
+    type: object
+    config:
+        fields:
+            baz: {type: String!}
+
+InterfaceA:
+    type: interface
+    config:
+        resolveType: '@=...'
+        fields:
+            bar: {type: String!}
+            bar2: {type: String!}
+            bar3: {type: String!}
+```
+
+This equivalent to:
+
+```yaml
+ObjectB:
+    type: object
+    config:
+        interfaces: [InterfaceA]
+        fields:
+             foo: {type: String!}
+             bar: {type: String!}
+             bar2: {type: String!}
+             bar3: {type: String!}
+             baz: {type: String!}
+
+ObjectA:
+    type: object
+    config:
+        fields:
+            bar: {type: String!}
+            bar2: {type: String!}
+            bar3: {type: String!}
+            foo: {type: String!}
+
+ObjectC:
+    type: object
+    config:
+        fields:
+            baz: {type: String!}
+
+InterfaceA:
+    type: interface
+    config:
+        resolveType: '@=...'
+        fields:
+            bar: {type: String!}
+            bar2: {type: String!}
+            bar3: {type: String!}
+```
+
+**Notes:**
+ * Inheritance works only with the configuration right now
```

Comment:
maybe?
```diff
- with the
+ with the yaml
```

Reply:
Inheritance works only with the generated types right now